replica/table.cc

```diff
@@ -2040,6 +2041,27 @@ future<db::replay_position> table::discard_sstables(db_clock::time_point truncat
     co_return p->rp;
 }
 
+void table::mark_ready_for_writes(db::commitlog* cl) {
+    if (!_readonly) {
+        on_internal_error(dblog, ::format("table {}.{} is already writable", _schema->ks_name(), _schema->cf_name()));
+    }
+    update_sstables_known_generation(sstables::generation_from_value(0));
+    if (_config.enable_commitlog) {
+        _commitlog = cl;
+    }
+    _readonly = false;
+}
+
+db::commitlog* table::commitlog() const {
+    if (_virtual_writer) [[unlikely]] {
+        return nullptr;
+    }
+    if (_readonly) [[unlikely]] {
```

Comment:
Hm, instead of two `if`s, I think it would be better to use `if (_readonly && !_virtual_writer)` -- for regular tables in normal operation this will be a single check.

(`return _commitlog` already returns `nullptr` for `_virtual_writer`s, right?)

(I'm worried about minimizing checks here because IIUC `commitlog()` is on the hot path, called for every write (?))